packages/next/build/webpack/loaders/eslint-loader/base-rules.ts

```diff
@@ -0,0 +1,173 @@
+  'no-unreachable': 'warn',
+  'no-unused-expressions': [
+    'error',
+    {
+      allowShortCircuit: true,
+      allowTernary: true,
+      allowTaggedTemplates: true,
+    },
+  ],
+  'no-unused-labels': 'warn',
+  'no-unused-vars': [
+    'warn',
+    {
+      args: 'none',
+      ignoreRestSiblings: true,
+    },
+  ],
+  'no-use-before-define': [
+    'warn',
+    {
+      functions: false,
+      classes: false,
+      variables: false,
+    },
+  ],
+  'no-useless-computed-key': 'warn',
+  'no-useless-concat': 'warn',
+  'no-useless-constructor': 'warn',
+  'no-useless-escape': 'warn',
+  'no-useless-rename': [
+    'warn',
+    {
+      ignoreDestructuring: false,
+      ignoreImport: false,
+      ignoreExport: false,
+    },
+  ],
+  'no-with': 'warn',
+  'no-whitespace-before-property': 'warn',
+  'require-yield': 'warn',
+  'rest-spread-spacing': ['warn', 'never'],
+  strict: ['warn', 'never'],
+  'unicode-bom': ['warn', 'never'],
+  'use-isnan': 'warn',
+  'valid-typeof': 'warn',
+  'no-restricted-properties': [
+    'error',
+    {
+      object: 'require',
+      property: 'ensure',
+      message:
+        'Please use import() instead. More info: https://facebook.github.io/create-react-app/docs/code-splitting',
+    },
+    {
+      object: 'System',
+      property: 'import',
+      message:
+        'Please use import() instead. More info: https://facebook.github.io/create-react-app/docs/code-splitting',
+    },
+  ],
+  'getter-return': 'warn',
+  'react-hooks/rules-of-hooks': 'error',
+  'react-hooks/exhaustive-deps': 'off',
+  'import/no-unresolved': [2, { commonjs: true, amd: true }],
+}
+
+// Copied from https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/src/configs/eslint-recommended.ts
+const tsRules = {
+  'constructor-super': 'off', // ts(2335) & ts(2377)
+  'getter-return': 'off', // ts(2378)
+  'no-const-assign': 'off', // ts(2588)
+  'no-dupe-args': 'off', // ts(2300)
+  'no-dupe-class-members': 'off', // ts(2393) & ts(2300)
+  'no-dupe-keys': 'off', // ts(1117)
+  'no-func-assign': 'off', // ts(2539)
+  'no-import-assign': 'off', // ts(2539) & ts(2540)
+  'no-new-symbol': 'off', // ts(2588)
+  'no-obj-calls': 'off', // ts(2349)
+  'no-redeclare': 'off', // ts(2451)
+  'no-setter-return': 'off', // ts(2408)
+  'no-this-before-super': 'off', // ts(2376)
+  'no-undef': 'off', // ts(2304)
+  'no-unreachable': 'off', // ts(7027)
+  'no-unsafe-negation': 'off', // ts(2365) & ts(2360) & ts(2358)
+  'no-var': 'error', // ts transpiles let/const to var, so no need for vars any more
+  'prefer-const': 'error', // ts provides better types with const
+  'prefer-rest-params': 'error', // ts provides better types with rest args over arguments
+  'prefer-spread': 'error', // ts transpiles spread to apply, so no need for manual apply
+  'valid-typeof': 'off', // ts(2367),
+  'no-unused-vars': 'off',
+}
+
+export const getBaseRules = (isTypescript: Boolean) => {
```

Comment:
Are we able to show which external `.eslintrc` files are being used and which rules they are overriding similar to how we show `tsconfig.json` config loading [here](https://github.com/vercel/next.js/blob/19fed3f83b88c587cbac943ea34fbfbbb76161ff/packages/next/lib/verifyTypeScriptSetup.ts)?

Reply:
When do you think we would want to show such information? during dev/build?